lib/streamlit/error_util.py

```diff
@@ -42,15 +44,33 @@ def handle_uncaught_app_exception(e: BaseException) -> None:
     if the user has disabled client error details, we display a generic
     warning in the frontend instead.
     """
+    exc_type, exc_value, exc_traceback = sys.exc_info()
```

Comment:
two nits:
* these don't get used until the `else` case, so I think it'd be best to move this call there
* is it necessary to use `traceback.format_exc()` to get the stacktrace? or can the same info be found in `exc_traceback`? If `exc_traceback` is unused, then it'd be good to prefix its name with an underscore to denote that (it's also fine to just completely replace the variable name with an underscore as that's conventionally used to mark an unused variable).